### DISCUSSION TITLE
client code load twice

### DISCUSSION THREAD

**Discussion by zhujiageng** [vikejs/vike#1119] [Discussion] [answered]
### Description

when i run the example/layouts-react, i found the Page will render twice, when i use vite-plugin-ssr in my project, my page alse render twice.
this will lead to the useEffec excute twice

**Comment by brillout** [comment]
I don't understand your problem. It seems to be a user land thing though, see https://brillout.github.io/rules/#try.

**Comment by zhujiageng** [comment]
https://github.com/brillout/vite-plugin-ssr/assets/103102265/66401a32-58d0-4d71-8fbb-45ce897421af

this is not a user land thing, because all code is generate by vite-plugin-ssr.
step 1: npm init vite-plugin-ssr@latest and run pnpm install
step 2: pnpm run dev and open http://localhost:3000/ with chrome
step 3: open chrome devtool, add breakpoint in pages/about/index.page.tsx render func
step 4: click the about link, you will find the render function excute twice
@brillout

**Reply by brillout** [reply, marked as answer]
That's how React's strict mode works.

You can inspect the `render()` hook and observe that vite-plugin-ssr calls it only once.
